src/main/java/fi/aalto/cs/apluscourses/ui/AboutDialog.java

```diff
@@ -58,11 +83,52 @@ private static JPanel createAboutInnerContainer() {
     return linkLabel;
   }
 
+  @NotNull
+  private static LinkLabel<Object> createAPlusWebsiteLinkLabel() {
+    LinkLabel<Object> linkLabel = new LinkLabel<>(
+            getText("ui.aboutDialog.APlusWebsite"),
+            AllIcons.Ide.External_link_arrow,
+            (first, second) ->
+                    BrowserUtil.browse(A_PLUS_PAGE));
+    linkLabel.setIconTextGap(0);
+    linkLabel.setHorizontalTextPosition(SwingConstants.LEFT);
+
+    return linkLabel;
+  }
+
+  @NotNull
+  private static LinkLabel<Object> createGithubWebsiteLinkLabel() {
+    LinkLabel<Object> linkLabel = new LinkLabel<>(
+            getText("ui.aboutDialog.GithubWebsite"),
+            AllIcons.Ide.External_link_arrow,
+            (first, second) ->
+                    BrowserUtil.browse(GITHUB_PAGE));
+    linkLabel.setIconTextGap(0);
+    linkLabel.setHorizontalTextPosition(SwingConstants.LEFT);
+
+    return linkLabel;
+  }
+  @NotNull
+  private static JBLabel createVersionTextLabel() throws IOException {
+    JBLabel label = new JBLabel();
+    Properties properties = new Properties();
+    properties.load(AboutDialog.class.getResourceAsStream("/build-info.properties"));
```

Comment:
I wonder if it's possible to use the existing `BuildInfo` class here (`src/main/java/fi/aalto/cs/apluscourses/utils/BuildInfo.java`), something like this (untested):

```
Version version = BuildInfo.INSTANCE.version;
```

`BuildInfo` also takes care of the IOException, which would be nice, since this code wouldn't have to worry about it and the `throws` could be removed

Reply:
You're right, it works fine and there is no need to handle a possible IOException.